workshops/fabric-e2e-rag/workshop.md

```diff
@@ -0,0 +1,606 @@
+| Access to Azure OpenAI API | [Request access to Azure OpenAI](https://aka.ms/oaiapply) |
+| A Web browser        | [Get Microsoft Edge](https://www.microsoft.com/edge) |
+| Python knowledge | [Python for beginners](https://learn.microsoft.com/en-us/training/paths/beginner-python/) |
+
+
+---
+
+
+# Introduction
+
+Analyzing structured data has been an easy process for some time but the same cannot be said for unstructured data. Unstructured data, such as text, images, and videos, is more difficult to analyze and interpret. However, with the advent of advanced AI models, such as OpenAI's GPT-3 and GPT-4, it is now becoming easier to analyze and gain insights from unstructured data.
+
+An example of such analysis is the ability to query a document for specific information using natural language which is achievable though a combination of information retrieval and language generation.
+
+By leveraging the RAG (Retrieval-Augmented Generation) framework, you can create a powerful question-and-answering pipeline that uses a large language model (LLM) and you own data to generate responses. 
+
+The architecture of such an application is as shown below:
+
+![schema](assets/schema.png)
+
+To get an in-depth understanding of the RAG framework, refer to [this workshop](https://moaw.dev/workshop/gh:azure-samples/azure-openai-rag-workshop/base/docs/)
+
+
+---
+
+# Environment Setup
+To continue with this workshop, you'll need to create a Lakehouse in your Microsoft Fabric workspace and deploy the necessary resources in your Azure account. We'll detail the steps to do this below.
+
+## Create a Lakehouse
+To create a new Lakehouse in your Microsoft Fabric workspace, open the `Data Engineering workload` on the bottom left of the workspace and click on `Lakehouse`. Provide the name `rag_workshop` and click `Create`
+
+![lakehouse](assets/lakehouse.png)
+
+To learn more about Lakehouses in Microsoft Fabric, refer to [this resource](https://learn.microsoft.com/en-us/fabric/data-engineering/tutorial-build-lakehouse#create-a-lakehouse?WT.mc_id=data-114676-jndemenge)
+
+
+## Azure Setup
+
+To complete this workshop you'll need an Azure account. If you don't have one, you can create a [free account](https://azure.microsoft.com/en-gb/free/?WT.mc_id=data-0000-cxa) before you begin. 
+
+<div class="important" data-title="Important">
+
+> Ensure that the subscription you are using has the permissions to create and manage resources.
+</div>
+
+
+Once you have the Account ready Right-click or `Ctrl + Click` on the button below to open the Azure Portal in a new tab and begin deployment of the necessary resources.
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FMicrosoft%2Fmoaw%2Fmain%2Fworkshops%2Ffabric-e2e-rag%2Fassets%2Fmain.json)
```

Comment:
I got two errors, noting here in case we want to document for users who may run into them:

* I already had a free tier search, and theres currently a limit of 1, so I just edited the template to be standard. Workshop followers can also delete their existing search service if not in use.
* I got this error about Responsible AI:
This subscription cannot create CognitiveServices until you agree to Responsible AI terms for this resource. You can agree to Responsible AI terms by creating a resource through the Azure Portal then trying again. For more detail go to https://aka.ms/csrainotice
I was surprised since I deploy OpenAI daily, but never created a multi-resource OpenAI, I guess. So I had to make one in the Portal in order to accept the Responsible AI checkbox.